presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -753,4 +753,13 @@ default TableLayoutFilterCoverage getTableLayoutFilterCoverage(ConnectorTableLay
     {
         return NOT_APPLICABLE;
     }
+
+    /**
+     * Allows the connector to reject the table scan produced by the planner.
+     * <p>
+     * Connectors can choose to reject a query based on the table scan potentially being too expensive, for example
+     * if no filtering is done on a partition column.
+     * <p>
+     */
+    default void validateScan(ConnectorSession session, ConnectorTableHandle handle, Optional<ConnectorTableLayoutHandle> layout) {}
```

Comment:
not a big fan of adding a new interface because we have a new rule. Shall we use `ConnectorPlanOptimizer` instead?